src/main/java/pl/tomaszdziurko/jvm_bloggers/blog_posts/domain/BlogPost.java

```diff
@@ -95,4 +95,73 @@ public String getApprovalState() {
     public boolean isGoingInNewsletter(final LocalDateTime lastPublicationDate) {
         return publishedDate.isAfter(lastPublicationDate);
     }
+
+    public static BlogPostBuilder builder() {
```

Comment:
It won't set it to `null` anymore after you made the `uid` field final. You can safely replace this custom builder implementation back to `@Builder`.